src/services/LeafValueCoder.ts

```diff
@@ -74,4 +91,22 @@ export class LeafValueCoder {
 
     return new BigNumber(n || '0', n.startsWith('0x') ? 16 : 10).toString(16);
   };
+
+  static isIntValue = (label: string): boolean => label.startsWith(INT_PREFIX);
+
+  static intValueToHex = (n: BigNumber | number): string => {
+    if (n >= 0) {
+      return n.toString(16);
+    }
+
+    return maxUint224.plus(n).plus(1).toString(16);
+  };
+
+  static toInt = (n: BigNumber): number => {
```

Comment:
```suggestion
  static toInt = (n: BigNumber): bigint => {
```

number will not handle all required values, we will loose precision

Reply:
> left some comments, also we need few tests to be added here: https://github.com/umbrella-network/canvas/blob/main/test/integration/coders.test.ts